website/docs/language/functions/issensitive.mdx

```diff
@@ -0,0 +1,40 @@
+---
+sidebar_label: issensitive
+description: >-
+  The issensitive function returns a boolean saying whether or not a value is
+  marked as sensitive.
+---
+
+# `issensitive` Function
+
+`issensitive` takes any value and returns `true` if the value is marked as
+sensitive, and `false` otherwise.
+
+The `issensitive` function might be useful if you need to programmatically
+determine whether or not a value is sensitive, for example if you have a value
+containing both sensitive and non-sensitive values and you need to separate the
+two parts:
+
+```hcl
+variable "environment_variables" {
+  description = "A list of environment variables that may contain both sensitive and non-sensitive values"
+  type        = map(string)
+  sensitive   = true
+}
+
+locals {
+  sensitive_variables = [for key, value in var.environment_variables: key if issensitive(value)]
+  nonsensitive_variables = [for key, value in var.environment_variables: key if !issensitive(value)]
+}
+```
```

Comment:
This example seems can not construct a list with both sensitive and nonsensitive variables.I would recommend using this
```
locals {
  mixed_variables = tolist(["a", sensitive("b"), "c"])
}

locals {
  sensitive_variables = [for key, value in local.mixed_variables: key if issensitive(value)]
  nonsensitive_variables = [for key, value in local.mixed_variables: key if !issensitive(value)]
}

```

Reply:
I tested the example in a module and used the module like this:

```
module "test" {
  source = "./test"
  environment_variables = {
    "TEST_SENSITIVE" = sensitive("A")
    "TEST_NONSENSITIVE" = "B"
  }
}

output "sensitive" {
  value = module.test.sensitive_variables
}

output "nonsensitive" {
  value = module.test.nonsensitive_variables
}
```

Which produced the following output:

```
Changes to Outputs:
  + nonsensitive = [
      + "TEST_NONSENSITIVE",
    ]
  + sensitive    = [
      + "TEST_SENSITIVE",
    ]
```

But I can change the example if it's more clear that way 😄